src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -286,9 +307,12 @@ private function getTransitions($transitionName)
      */
     private function getTransitionForSubject($subject, Marking $marking, array $transitions)
     {
+        $strategy = $this->markingStore->getStrategy();
         foreach ($transitions as $transition) {
             foreach ($transition->getFroms() as $place) {
-                if (!$marking->has($place)) {
+                if ($marking->has($place) && Marking::STRATEGY_MULTIPLE_STATE === $strategy) {
```

Comment:
this change is wrong. If there are multiple from, we must check **all** of them to be sure that the marking has them (it is a AND between them, not a OR)